### DISCUSSION TITLE
open link directory

### DISCUSSION THREAD

**Discussion by Olifair** [freeplane/freeplane#619] [Ideas]
Perhaps just a small improvement, but one that I would use often (probably because I'm used to this from Thunderbird) : 
I use quite often links to files in the nodes, so that I can easily open the file with one click. 
But sometimes I would like to just open the directory in the system file viewer where the file is located (or have a possibility to copy the path to the directory). 
I know that I can link to a directory, but this would mean two nodes, one for the file and one for the directory ... 
Just a right click on the file link icon, with the option "Open link directory" would be magic :-)

Cheers

Oliver

**Comment by macmarrum** [comment]
Here's a scrip that opens the directory of a linked file
https://github.com/macmarrum/freeplane-scripting/blob/main/src/scripts/openDirectoryOfLinkedFile.groovy

Feel free to add a hot key for it.
And if you remove `="/menu_bar/Mac2"` from the first line of the script, it will be available in node-right-click `Scripts->Open Directory Of Linked File`

**Comment by Olifair** [comment]
Wow, wonderful, thanks a lot, this is great. 
And just for some others who find this thread : I just copy and pasted the script into a text file, with "openDirectoryOfLinkedFile.groovy" as name and copied it into the scripts folder of the Freeplane system folder. The original version gives a new menu called "Mac2", and when taking out  ="/menu_bar/Mac2"  one gets a right click menu.  Exactly what I have wished for
thanks again
